### PR TITLE
Withdraws

### PR DESCRIPTION
**Pull request by aaalata**
Withdraw money scenario.

### CODE REVIEW

File: src/main/java/pl/edu/agh/iet/katabank/Account.java [hunk 1]
```diff
@@ -34,7 +34,20 @@ public boolean equals(Object o) {
...
         return Objects.hash(id);
     }
+
+    public void setBalance(BigDecimal balance) {
+        this.balance = balance;
+    }
+
+    public void withdraw(BigDecimal amount) throws IllegalArgumentException {
+        if (amount.signum()<0){
+            throw new IllegalArgumentException("The amount to withdraw is negative.");
```
Comment (anthonyflea) [inline src/main/java/pl/edu/agh/iet/katabank/Account.java]: I think we should also check amount for null.
Throw exception when the amount equals 0 not only negative value. (Something to consider)
Print the amount value in the exception message.

**Review by anthonyflea — CHANGES_REQUESTED** [review]

File: src/test/groovy/pl/edu/agh/iet/katabank/AccountTest.groovy [hunk 2]
```diff
@@ -16,4 +16,25 @@ class AccountTest extends Specification{
         where:
         anotherAccount = new Account(customer)
     }
+
+    def "try to withdraw a negative amount"() {
+        when:
+        account.withdraw(new BigDecimal(-1))
```
Comment (anthonyflea) [inline src/test/groovy/pl/edu/agh/iet/katabank/AccountTest.groovy]: In groovy java.math.BigDecimal is the default type for non-integer floating types. You don't have to create them using the new keyword.

**Review by anthonyflea — CHANGES_REQUESTED** [review]

File: src/test/groovy/pl/edu/agh/iet/katabank/AccountTest.groovy (hunk 2 above)
Comment (aaalata) [reply src/test/groovy/pl/edu/agh/iet/katabank/AccountTest.groovy]: Corrected in all test cases.

File: src/main/java/pl/edu/agh/iet/katabank/Account.java (hunk 1 above)
Comment (aaalata) [reply src/main/java/pl/edu/agh/iet/katabank/Account.java]: Code corrected.

**Comment by aaalata** [comment]
Code corrected.

**Review by anthonyflea — APPROVED** [review]

**Comment by anthonyflea** [comment]
Code review finished. Branches merged.

**Pull request merged**